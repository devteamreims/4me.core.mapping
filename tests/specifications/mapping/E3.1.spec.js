import app from '../../../index';
import request from 'supertest';

import fp from 'lodash/fp';

jest.mock('../../../config/cwps', () => {
  const mockCwps = [{id: 1, type: "cwp"}, {id: 2, type: "cwp"}];
  return mockCwps;
});

jest.mock('../../../config/sectors', () => {
  const mockSectors = [
    {name: "UR", elementarySectors: ["UR"]},
    {name: "XR", "elementarySectors": ["XR"]},
  ];
  return mockSectors;
});

describe('E3.1 : software integrates in 4ME framework', () => {

  test('has a /status endpoint', () => {
    return request(app)
      .get('/status')
      .expect(200)
      .expect(res => expect(res.body.version).toBe(process.env.npm_package_version));
  });

  test('produce log on new room configuration', () => {
    const newMap = [
      {cwpId: 2, sectors: ["UR", "XR"]},
    ];

    return request(app)
      .post('/mapping')
      .send(newMap)
      .expect(res => {
        const logRecord = fp.pipe(
          fp.get('LOG_STREAM.records'),
          fp.last,
          fp.pick(['map', 'msg', "name", "payload"])
        )(global);

        expect(logRecord).toMatchSnapshot();
      });
  });


});


