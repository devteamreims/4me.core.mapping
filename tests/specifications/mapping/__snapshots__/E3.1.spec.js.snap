exports[`E3.1 : software integrates in 4ME framework produce log on new room configuration 1`] = `
Object {
  "map": Array [
    Object {
      "cwpId": 2,
      "sectors": Array [
        "UR",
        "XR"
      ]
    }
  ],
  "msg": "",
  "name": "mapping-ops",
  "payload": Object {
    "commitMap": true
  }
}
`;
